### ISSUE TITLE
Roslyn fails to update state of a member dereferenced via property initializer to non-null

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#38062]
**Version Used**: 

```
Branch master (13 Aug 2019)
Latest commit a08336 by David:
Merge pull request #37916 from dibarbet/partial_load_str

Update partial load dialog string to better match task center.
```

**Steps to Reproduce**:

Compile and run the following code:
```cs
#nullable enable
class C
{
  C? f;
    
  static void Main()
  {
    var c = new C() { f = { f = null }}; // CS8602, correct
    c.f.ToString(); // CS8602, incorrect
  }
}
```

**Expected Behavior**:
No warnings for `c.f.ToString()` 

**Actual Behavior**:
`warning CS8602: Dereference of a possibly null reference.` is reported for `c.f.ToString()`.

**Notes**
The code above is equivalent to the following:
```
var c = new C();
c.f.f = null; // CS8602, correct
c.f.ToString();
```
in this example however Roslyn correctly updates the field's state to non-null but fails to do so if object initializer syntax is used.

**Comment by TessenR** [comment]
Sorry, it's a duplicate of https://github.com/dotnet/roslyn/issues/34868
